confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -0,0 +1,588 @@
+
+        self._rest_client.POST(
+            url_formatter(VERSIONS, subject_name),
+            body=schema, encoder=_SchemaJSONEncoder)
+
+        # get registration information
+        return self._rest_client.POST(
+            url_formatter(SUBJECT, subject_name),
+            body=schema, encoder=_SchemaJSONEncoder, decoder=_VersionJSONDecoder,
+            schema=schema)
+
+    def get_schema(self, schema_id):
+        """
+        GET /schemas/ids/{int: schema_id}
+
+        Fetches the schema associated with ``schema_id`` from the
+        Schema Registry. The result is cached so subsequent attempts will not
+        require an additional trip to the Schema Registry.
+
+        Args:
+            - schema_id (int): Schema id
+
+        Returns:
+            Schema: Schema string identified by the ``schema_id``
+
+        Raises:
+            SchemaRegistryClientError: If schema can't be found
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#get--schemas-ids-int-%20id
+
+        """  # noqa: E501
+        version = self.id_to_schema.get(schema_id, None)
+
+        if version is not None:
+            return version
+
+        url = url_formatter(SCHEMAS, schema_id)
+        schema = self._rest_client.GET(url, decoder=_SchemaJSONDecoder)
+        self.id_to_schema[schema_id] = schema
+
+        return schema
+
+    def get_registration(self, subject_name, schema):
+        """
+        POST /subjects/(string: subject)
+
+        Returns ``schema`` registration information for ``subject``.
+
+        Args:
+            - subject_name (str): Subject name the schema is registered under
+            - schema (Schema): Registered Schema
+
+        Returns:
+            Version: ``schema`` registration information for ``subject``
+
+        Raises:
+            SchemaRegistryClientError: If schema or subject can't be found
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#post--subjects-(string-%20subject)
+
+        """  # noqa: E501
+        version = schema.subjects.get(subject_name, None)
+        if version:
+            return version
+
+        return self._rest_client.POST(
+            url_formatter(SUBJECT, subject_name),
+            encoder=_SchemaJSONEncoder,
+            decoder=_VersionJSONDecoder,
+            schema=schema)
+
+    def get_subjects(self):
+        """
+        GET /subjects
+
+        List all subjects registered with the Schema Registry
+
+        Returns:
+            [str]: Registered subject names
+
+        Raises:
+            SchemaRegistryClientError: if subjects can't be found
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#get--subjects
+
+        """  # noqa: E501
+        return self._rest_client.GET(url_formatter(SUBJECTS))
+
+    def delete_subject(self, subject):
```

Comment:
in .net i haven't even bothered with methods yet that aren't used by the serdes, but nice to have.